### PR TITLE
Business Summary end point + basic framework + summary url in busines…

### PR DESCRIPTION
**Pull request by lekshmimallika-aot**
…s object

*Issue #:* /bcgov/entity#4463

*Description of changes:*
Include business summary url in the business json
New endpoint to download business summary
A new class for processing business summary as the current report class mainly deals with filings and already has lot of conditions to handle reports for different filing types.
Test updates

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Review by vysakh-menon-aot — APPROVED** [review]

File: legal-api/src/legal_api/models/business.py [hunk 1]
```diff
@@ -335,9 +336,8 @@ def json(self):
         if self.tax_id:
             d['taxId'] = self.tax_id
         if self.state_filing_id:
-            base_url = current_app.config.get('LEGAL_API_BASE_URL')
             d['stateFiling'] = f'{base_url}/{self.identifier}/filings/{self.state_filing_id}'
-
+        d['summary'] = f'{base_url}/{self.identifier}/summary'
```
Comment (thorwolpert) [inline legal-api/src/legal_api/models/business.py:340]: should the summary be in the documents listing?
would it be an alternate view of the /businesses/identifier?output=summary as the default output would be json?

**Review by thorwolpert — COMMENTED** [review]
question on URL pattern layout

File: legal-api/src/legal_api/models/business.py (hunk 1 above)
Comment (lekshmimallika-aot) [reply legal-api/src/legal_api/models/business.py:340]: It need not be. The summary had  more information like all the alterations that happened previously. We dont get from any of the existing endpoints. All other info can be retrieved from existing endpoints.

I will move the code in the new endpoint to the existing businesses endpoint and delete the new endpoint + remove thhe url from the business json

File: legal-api/report-templates/template-parts/common/businessDetails.html [hunk 2]
```diff
@@ -1,7 +1,30 @@
 <div class="business-details-container">
    <table class="business-details-table" role="presentation">
       <tr>
-         {% if header.reportType == 'noticeOfArticles' %}
+         {% if reportType == 'businessSummary' %}
```
Comment (argush3) [inline legal-api/report-templates/template-parts/common/businessDetails.html:4]: should this be `header.reportType` as opposed to `reportType`?

Comment (argush3) [reply legal-api/report-templates/template-parts/common/businessDetails.html:4]: nvm, i see where this comes from(`BusinessSummary._get_template_data` function) now.

**Review by argush3 — APPROVED** [review]

**Review by thorwolpert — APPROVED** [review]

**Pull request merged**